src/errors.rs

```diff
@@ -0,0 +1,36 @@
+use reqwest::Error as ReqwestError;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub enum TurnkeyError {
+    MethodError(TurnkeyResponseError),
+    StampError(StampError),
+    HttpError(ReqwestError),
+    OtherError(String),
```

Comment:
This could likely be replaced by `anyhow` to avoid the boilerplate related to string conversions. It would then also allow for specifying the context. `e.g. result.context("failed to do x")`.